gwcs/coordinate_frames.py

```diff
@@ -9,53 +9,57 @@
 from astropy import units as u
 from astropy import utils as astutil
 from astropy import coordinates as coord
-from astropy.coordinates import (BaseCoordinateFrame, FrameAttribute,
-                                 RepresentationMapping)
+from astropy.extern import six
 
-from . import spectral_builtin_frames
-from .spectral_builtin_frames import *
-from .representation import *
+from . import utils as gwutils
 
 
-__all__ = ['DetectorFrame', 'CelestialFrame', 'SpectralFrame',
-           'CompositeFrame', 'FocalPlaneFrame']
+__all__ = ['Frame2D', 'CelestialFrame', 'SpectralFrame', 'CompositeFrame',
+           'CoordinateFrame']
 
 
-STANDARD_REFERENCE_POSITION = ["GEOCENTER", "BARYCENTER", "HELIOCENTER",
-                               "TOPOCENTER", "LSR", "LSRK", "LSRD",
-                               "GALACTIC_CENTER", "MOON", "LOCAL_GROUP_CENTER"]
+##STANDARD_REFERENCE_POSITION = ["GEOCENTER", "BARYCENTER", "HELIOCENTER",
+##"TOPOCENTER", "LSR", "LSRK", "LSRD",
+##"GALACTIC_CENTER", "MOON", "LOCAL_GROUP_CENTER"]
 
 
 class CoordinateFrame(object):
 
     """
-    Base class for CoordinateFrames
+    Base class for CoordinateFrames.
 
     Parameters
     ----------
     naxes : int
         Number of axes.
-    axes_order : tuple
-        A mapping of inputs to coordinate frame axes order.
-    reference_frame : astropy.coordinates.builtin_frames or spectral_builtin_frames
-        Reference frame (see subclasses).
-    reference_position : str or tuple
-        Reference position - one of `STANDARD_REFERENCE_POSITION` or a tuple of floats
+    axes_type : str
+        One of ["SPATIAL", "SPECTRAL", "TIME"]
```

Comment:
I think FITS WCS documents use the word "CELESTIAL" in place of "SPATIAL" here.  (Or maybe I just misunderstand what's happening here).  I don't have strong feelings -- I actually think "SPATIAL" seems more general -- but it was a bit surprising to me.